src/applications/appeals/995/validations/date.js

```diff
@@ -1,54 +1,46 @@
 import moment from 'moment';
 
-import { parseISODate } from 'platform/forms-system/src/js/helpers';
-
-import { fixDateFormat } from '../../shared/utils/replace';
-import { errorMessages, FORMAT_YMD, MAX_YEARS_PAST } from '../constants';
+import { errorMessages, MAX_YEARS_PAST } from '../constants';
 
 import {
-  buildErrorParts,
-  createPartsError,
-  isInvalidDateString,
-  hasErrorParts,
+  createScreenReaderErrorMsg,
+  dateFunctions,
 } from '../../shared/validations/date';
 
 export const minDate = moment()
   .subtract(MAX_YEARS_PAST, 'year')
   .startOf('day');
 
-const maxDate = moment().startOf('day');
-
 export const validateDate = (errors, rawString = '', fullData) => {
-  const dateString = fixDateFormat(rawString);
-  const { day, month, year } = parseISODate(dateString);
-  const date = moment(rawString, FORMAT_YMD);
-  // get last day of the month (month is zero based, so we're +1 month, day 0);
-  // new Date() will recalculate and go back to last day of the previous month
-  const maxDays = year && month ? new Date(year, month, 0).getDate() : 31;
+  const {
+    datePartErrors,
+    isInvalidDateString,
+    hasErrorDate,
+    date,
+    todayOrFutureDate,
+  } = dateFunctions(rawString);
   const dateType = fullData?.dateType || 'decisions';
-  const invalidDate = dateString?.length < FORMAT_YMD.length || !date.isValid();
-  const errorParts = buildErrorParts(month, day, year, maxDays);
 
-  if (isInvalidDateString(year, day, month, invalidDate)) {
+  if (isInvalidDateString) {
```

Comment:
I think we can replace all of these comparsons with `dateErrorMsgs` once we figure out which error messages to use.

Reply:
Yes, once content has been updated by design, we'll be able to consolidate the error messages to be shared across all 3